From 612ac829ed636d331c95f9370e8a1f26aac0354d Mon Sep 17 00:00:00 2001
From: Alan Modra <amodra@gmail.com>
Date: Thu, 6 Jun 2013 14:48:22 +0930
Subject: [PATCH] powerpc*le-linux support

On Thu, Jun 06, 2013 at 11:31:34AM +0930, Alan Modra wrote:
> This adds support for little-endian powerpc linux, and tidies the
> existing host match for powerpc.  config.sub won't return ppc*-*linux*
> so there isn't much point in matching that.

> -       ppc*-*linux*|powerpc*-*linux*)
> +       powerpcle*)
> +         LD="${LD-ld} -m elf64lppc"
> +         ;;
> +       powerpc*)
>           LD="${LD-ld} -m elf64ppc"
>           ;;

I didn't get that quite right.  'powerpc*' in the above matches too
much, for example when your host is powerpc64-linux and target
powerpc64le-linux you'll get -melf64ppc added to LD.  Since
powerpc64le-linux-ld wants -melf64lppc (or nothing) that will fail.
Revised as follows.

        * m4/libtool.m4 (ld -m flags): Remove non-canonical ppc host match.
        Support little-endian powerpc linux host.
---
 libltdl/m4/libtool.m4 |   12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

Index: libtool-2.4.2/libltdl/m4/libtool.m4
===================================================================
--- libtool-2.4.2.orig/libltdl/m4/libtool.m4	2013-07-05 13:51:03.143935985 +0800
+++ libtool-2.4.2/libltdl/m4/libtool.m4	2013-07-05 13:51:05.655943825 +0800
@@ -1312,7 +1312,7 @@
   rm -rf conftest*
   ;;
 
-x86_64-*kfreebsd*-gnu|x86_64-*linux*|ppc*-*linux*|powerpc*-*linux*| \
+x86_64-*kfreebsd*-gnu|x86_64-*linux*|powerpc*-*linux*| \
 s390*-*linux*|s390*-*tpf*|sparc*-*linux*)
   # Find out which ABI we are using.
   echo 'int i;' > conftest.$ac_ext
@@ -1333,7 +1333,10 @@
 		;;
 	    esac
 	    ;;
-	  ppc64-*linux*|powerpc64-*linux*)
+	  powerpc64le-*)
+	    LD="${LD-ld} -m elf32lppclinux"
+	    ;;
+	  powerpc64-*)
 	    LD="${LD-ld} -m elf32ppclinux"
 	    ;;
 	  s390x-*linux*)
@@ -1352,7 +1355,10 @@
 	  x86_64-*linux*)
 	    LD="${LD-ld} -m elf_x86_64"
 	    ;;
-	  ppc*-*linux*|powerpc*-*linux*)
+	  powerpcle-*)
+	    LD="${LD-ld} -m elf64lppc"
+	    ;;
+	  powerpc-*)
 	    LD="${LD-ld} -m elf64ppc"
 	    ;;
 	  s390*-*linux*|s390*-*tpf*)
Index: libtool-2.4.2/libltdl/config/config.guess
===================================================================
--- libtool-2.4.2.orig/libltdl/config/config.guess	2013-07-05 13:51:03.143935985 +0800
+++ libtool-2.4.2/libltdl/config/config.guess	2013-07-05 13:51:05.655943825 +0800
@@ -967,6 +967,12 @@
     ppc:Linux:*:*)
 	echo powerpc-unknown-linux-gnu
 	exit ;;
+    ppc64le:Linux:*:*)
+	echo powerpc64le-unknown-linux-${LIBC}
+	exit ;;
+    ppcle:Linux:*:*)
+	echo powerpcle-unknown-linux-${LIBC}
+	exit ;;
     s390:Linux:*:* | s390x:Linux:*:*)
 	echo ${UNAME_MACHINE}-ibm-linux
 	exit ;;
